sources/ghostdog/src/main.rs

```diff
@@ -160,6 +179,40 @@ fn is_neuron_attached() -> Result<()> {
     }
 }
 
+/// Detects if infiniband is present. If not, return early. If it does find Infiniband devices,
+/// find if they match specific capabilities to be used for communication between NVIDIA Fabric
+/// Manger and NVLSM, then write the guid to an env file for use by those services.
+fn find_and_write_infiniband_guid(env_file: PathBuf) -> Result<()> {
+    let devices = find_infiniband_devices()?;
+
+    // Return early if no devices are found
+    if devices.is_empty() {
+        return Ok(());
+    }
+
+    // For each device, confirm if SW_MNG is present, then find the first port of the device. If that
+    // device has the correct capability mask, then get the GUID. The first device to be found in this
+    // search is the correct GUID for the configuration file.
+    for device in devices {
+        if device.is_device_sw_mng()? {
+            let ports = device.find_ports_for_device()?;
+            for port in ports {
+                if port.is_sm_enabled() {
+                    // NVIDIA Fabric Manager or NVLSM use -g to configure a Port GUID, use this as input
+                    // to those services
+                    write_config_string(
+                        env_file.clone(),
+                        format!("GUID_ARG=\"-g {}\"", *port.port_guid).as_str(),
```

Comment:
nit: this subcommand could take `env_var` and `env_val_prefix` arguments to avoid hard-coding this logic